File: solutions/typescript/line-up/1/line-up.ts
export function format(name: string, number: number): string {
  const lastTwo = number % 100;
  const lastOne = number % 10;

  let ending: string;

  if (lastTwo >= 11 && lastTwo <= 13) {
    ending = "th";
  } else if (lastOne === 1) {
    ending = "st";
  } else if (lastOne === 2) {
    ending = "nd";
  } else if (lastOne === 3) {
    ending = "rd";
  } else {
    ending = "th";
  }

  return `${name}, you are the ${number}${ending} customer we serve today. Thank you!`;
}

File: solutions/csharp/line-up/1/LineUp.cs
public static class LineUp
{
    public static string Format(string name, int number)
    {
        return $"{name}, you are the {number}{GetOrdinal(number)} customer we serve today. Thank you!";
    }

    public static string GetOrdinal(int number)
    {
        int lastTwoDigits = Math.Abs(number % 100);
        if (lastTwoDigits != 11 && lastTwoDigits != 12 && lastTwoDigits != 13)
        {
            if (number % 10 == 1) return "st";
            if (number % 10 == 2) return "nd";
            if (number % 10 == 3) return "rd";
        }
        return "th";
    }
}
